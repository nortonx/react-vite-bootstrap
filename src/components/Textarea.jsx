import PropTypes from "prop-types";
import "../scss/Textarea.scss";
const Textarea = ({ label, id, onChangeText }) => {
  return(
    <div className="textarea mb-3">
      <label htmlFor={id} className="form-label">{label}</label>
      <textarea 
        className="form-control" 
        id={id}
        onChange={(e) => onChangeText(e.target.value)}
        data-testid="textarea"
      ></textarea>
    </div>
  );
};

Textarea.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
  onChangeText: PropTypes.func
};

export default Textarea;