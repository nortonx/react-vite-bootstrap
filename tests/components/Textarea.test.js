import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Textarea from "../../src/components/Textarea";

describe("Textarea", () => {
  
  it("Should render component and match snapshot", async () => {
    const onChangeText = jest.fn();
    render(<Textarea label="New Label" onChangeText={onChangeText} />);
    const textarea = await screen.findByTestId("textarea");
    expect(textarea).toBeInTheDocument();
    expect(textarea).toMatchSnapshot();
  });

  it("Should fire event when text is changed", async () => {
    const onChangeText = jest.fn();
    render(<Textarea label="New Label" onChangeText={onChangeText} />);
    const textarea = await screen.findByTestId("textarea");
    fireEvent.change(textarea, { target: {value: "Lorem ipsum"}})
    expect(onChangeText).toHaveBeenCalledTimes(1)
  })
})