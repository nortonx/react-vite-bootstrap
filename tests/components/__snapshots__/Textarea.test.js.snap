// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Should render component and match snapshot 1`] = `
<textarea
  class="form-control"
  data-testid="textarea"
/>
`;
